### DISCUSSION TITLE
Dark themes dependencies

### DISCUSSION THREAD

**Discussion by domichel** [helge17/tuxguitar#556] [General]
Into a docker with
```
FROM debian:sid
RUN apt-get update && \
    env DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
      tuxguitar \
      tuxguitar-alsa \
    && rm -rf /var/lib/apt/lists/*
CMD tuxguitar
```
The dark themes are not working well. The colours doesn't change and most of the icons on the left panel are missing.

I try to build it with the recommends, but that didn't solved that issue.
What are the needed dependencies?

**Comment by guiv42** [comment]
I don't think there is any dependency issue. All that TuxGuitar does when changing skin is to update the icons.

"The colors don't change": I guess you are talking about the background color of widgets (toolbars, menus, dialogs, etc). This is normal. Background color is set by the OS/Desktop, not by TuxGuitar. Changing skin in TuxGuitar has no effect on background color. I don't know which desktop you are using, on my side I'm using KDE, and if I set the global KDE configuration to dark mode then background of TuxGuitar widgets becomes dark as expected. Notice that it's necessary to quit/restart TuxGuitar for the background color to be updated.

"most of the icons are missing": I don't think they are "missing", they may just be set correctly by the skin. That means a quite bright foreground. If you background did not change, they are just barely visible (see screenshots in [this post](https://github.com/helge17/tuxguitar/issues/27#issuecomment-1538648083))

**Comment by domichel** [comment]
In my regular installation, I run fvwm with the fvwm-crystal desktop. And use lxappearance and qt5ct to set the toolkits themes. I can try to add them into the docker image along with adwaita-dark and  fvwm.

I am trying tuxguitar into a docker because it depend on webkit-gtk, which is a calamity to compile on gentoo, both in terms of time and memory/cpu usage, and tuxguitar is the only installed software that depend on it.

**Reply by guiv42** [reply]
I think wekbit-gtk is only used to display user help (these are html files), TuxGuitar may as well work without this dependency. I don't remember exactly, but it may just try to open help with system browser if it can't access webkit. Not completely sure, but it may be worth a try.
Display of help pages *within* the app and not through system browser was introduced to bypass Ubuntu/snap limitation: default Firefox browser cannot acces local filesystem

**Reply by guiv42** [reply]
Just checked in the code: I confirm TuxGuitar tries to open help files with local browser if it fails to access webkit.
If you want to remove this dependency at build time, you will need to patch source. A possible way to do this:
- delete [SWTBrowser](https://github.com/helge17/tuxguitar/blob/master/desktop/TuxGuitar-ui-toolkit-swt/src/org/herac/tuxguitar/ui/swt/widget/SWTBrowser.java) class
- patch [SWTFactory.createBrowser()](https://github.com/helge17/tuxguitar/blob/537c93fd4ee435bfbaa1dfe9e5107dcd135162a4/desktop/TuxGuitar-ui-toolkit-swt/src/org/herac/tuxguitar/ui/swt/SWTFactory.java#L384), just return `null`

**Reply by domichel** [reply]
I just added a html USE flag that disable/enable the webkit-gtk depend, and it work fine without it. The help open in Firefox. I will check your patch suggestion later.

**Reply by guiv42** [reply]
Good news. I did not even think it could work without the patch, I think you can just forget it!

**Reply by domichel** [reply]
Well, You are right, tuxguitar depend on swt anyway. And I can also forget this docker.

**Comment by domichel** [comment]
I started a discussion on the gentoo's forum. gumbo is a light-weight alternative to webkit-gtk: https://codeberg.org/gumbo-parser/gumbo-parser

claws-mail can use either gtk-webkit or gumbo.
